### PR TITLE
Feature/enhancing-employee-list-and-filtering

### PR DESCRIPTION
**Pull request by mzakariia**

### CODE REVIEW

File: invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/controllers/UserController.java [hunk 1]
```diff
@@ -42,4 +49,32 @@ public ResponseEntity<User> getUser(@RequestParam long ID){
...
+        } else if (type.equals("allowedBalance")) {
+            List<Integer> intList = new ArrayList<Integer>();
+            for (String s : values) intList.add(Integer.parseInt(s));
+            return ResponseEntity.ok(filterUserService.filterEmployeeByBalance(intList));
+        } else if (type.equals("remainingBalance")) {
+            List<Integer> intList = new ArrayList<Integer>();
+            for (String s : values) intList.add(Integer.parseInt(s));
+            return ResponseEntity.ok(filterUserService.filterEmployeeByRemainBalance(intList));
+        }
+        return null;
```
Comment (MOSTA3D) [inline invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/controllers/UserController.java:77]: to reduce the amount of code, try declare a variable to hold the , and return one only one ResponseEntity

File: invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/dtos/TeamDetails.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (MOSTA3D) [inline invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/dtos/TeamDetails.java:11]: you can use annotation `@Data` instead

File: invoice-tracker-backend/src/main/resources/application.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ spring:
     driverClassName: com.mysql.cj.jdbc.Driver
     url: jdbc:mysql://localhost:3306/employeedb
     username: root
-    password: 123123123
+    password: ZakariiA4758!
```
Comment (MOSTA3D) [inline invoice-tracker-backend/src/main/resources/application.yml:13]: to reduce these conflicts we can use environment variables here too

File: invoice-tracker-frontend/src/pages/EmployeesHub/EmployeesHub.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        .then((response) => {
+          setEmployeeData(response.data);
+        });
+    };
+    getData();
+  }, []);
+
+  return (
+    <div>
+      <Navbar />
```
Comment (MOSTA3D) [inline invoice-tracker-frontend/src/pages/EmployeesHub/EmployeesHub.tsx:189]: Navbar shouldn't be here i guess, it should exist on a higher level like level of `Routes`, as you will end up with many navbars, each one with its own state.

File: invoice-tracker-frontend/src/utils/FetchFacad.ts [hunk 5]
```diff
@@ -1,7 +1,10 @@
 // handling get and post requests
 
-
-export class FetchFacad{
+// async function som3aFunc(){
```
Comment (MOSTA3D) [inline invoice-tracker-frontend/src/utils/FetchFacad.ts:3]: you forgot som3a here

**Review by MOSTA3D — APPROVED** [review]

File: invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/controllers/UserController.java [hunk 6]
```diff
@@ -42,4 +49,32 @@ public ResponseEntity<User> getUser(@RequestParam long ID){
         return ResponseEntity.created(uri).body(userService.saveRole(role));
     }
 
+    @GetMapping("/users/filter")
+    public ResponseEntity<List<User>> filterEmployees(@RequestParam("type") String type, @RequestParam("values") List<String> values) {
+
+
+        if (type.equals("englishName")) {
```
Comment (Abdelrhman-Samir-99) [inline invoice-tracker-backend/src/main/java/net/talaatharb/invoicetracker/controllers/UserController.java:56]: Consider using switch case for code clarity.

**Review by Abdelrhman-Samir-99 — APPROVED** [review]

**Review by hashemy21 — APPROVED** [review]

**Pull request merged**